src/Eryph.ConfigModel.Catlets.Validation/CatletConfigValidations.cs

```diff
@@ -0,0 +1,73 @@
+        | ValidateProperty(toValidate, c => c.Location, StorageIdentifier.NewValidation, path)
+        | ValidateList(toValidate, c => c.Drives, ValidateCatletDriveConfig, path, minCount: 0, maxCount: 64)
+        | ValidateProperty(toValidate, c => c.Cpu, ValidateCatletCpuConfig, path)
+        | ValidateProperty(toValidate, c => c.Memory, ValidateCatletMemoryConfig, path)
+        | ValidateList(toValidate, c => c.Fodder, ValidateCatletFodderConfig, path);
+
+    public static Validation<ValidationIssue, Unit> ValidateCatletDriveConfig(
+        CatletDriveConfig toValidate,
+        string path = "") =>
+        ValidateProperty(toValidate, c => c.Name, CatletDriveName.NewValidation, path)
+        | ValidateProperty(toValidate, c => c.Store, DataStoreName.NewValidation, path)
+        | ValidateProperty(toValidate, c => c.Location, StorageIdentifier.NewValidation, path)
+        | ValidateProperty(toValidate, c => c.Size, ValidateCatletDriveSize, path)
+        | ValidateProperty(toValidate, c => c.Source, source =>
+            source.StartsWith("gene:")
+                ? GeneIdentifier.NewValidation(source).Map(_ => unit)
+                : Validations.ValidatePath(source, "source").Map(_ => unit)
+                    .ToEither()
+                    .MapLeft(errors => Error.New("The source must be a valid gene identifier or path.", Error.Many(errors)))
+                    .ToValidation(),
+            path);
+    
+    private static Validation<Error, Unit> ValidateCatletDriveSize(int size) =>
+        guard(size > 0, Error.New("The drive size must be positive.")).ToValidation()
+        | guardnot(size > 64 * 1024, Error.New("The drive size must be at most 64 TB.")).ToValidation();
+
+    public static Validation<ValidationIssue, Unit> ValidateCatletCpuConfig(
+        CatletCpuConfig toValidate,
+        string path = "") =>
+        ValidateProperty(toValidate, c => c.Count, ValidateCatletCpuCount, path);
+
+    private static Validation<Error, Unit> ValidateCatletCpuCount(int count) =>
+        guard(count > 0, Error.New("The number of CPUs must be positive.")).ToValidation()
+        | guardnot(count > 240, Error.New("The number of CPUs must be at most 240.")).ToValidation();
+
+    public static Validation<ValidationIssue, Unit> ValidateCatletMemoryConfig(
+        CatletMemoryConfig toValidate,
+        string path = "") =>
+        ValidateProperty(toValidate, c => c.Maximum, ValidateCatletMemorySize, path)
+        | ValidateProperty(toValidate, c => c.Minimum, ValidateCatletMemorySize, path)
+        | ValidateProperty(toValidate, c => c.Startup, ValidateCatletMemorySize, path);
+
+    private static Validation<Error, Unit> ValidateCatletMemorySize(int memorySize) =>
+        guard(memorySize >= 32, Error.New("The memory size must be least 32 MB.")).ToValidation()
+        | guard(memorySize % 2 == 0, Error.New("The memory size must be a multiple of 2 MB.")).ToValidation()
```

Comment:
maybe we should also add this as requirement: https://learn.microsoft.com/en-us/windows-server/virtualization/hyper-v/supported-ubuntu-virtual-machines-on-hyper-v

If you're using Dynamic Memory on Windows Server 2019, Windows Server 2016 or Windows Server 2012/2012 R2 operating systems, specify Startup memory, Minimum memory, and Maximum memory parameters in multiples of 128 megabytes (MB). Failure to do so can lead to Hot-Add failures, and you might not see any memory increase on a guest operating system.